hazelcast/src/test/java/com/hazelcast/internal/dynamicconfig/DynamicConfigTest.java

```diff
@@ -148,8 +151,8 @@ public void testMultiMapConfig_whenEntryListenerConfigHasImplementation() {
 
     @Test
     public void testCardinalityEstimatorConfig() {
-        CardinalityEstimatorConfig config = new CardinalityEstimatorConfig(name, 4, 2);
-
+        CardinalityEstimatorConfig config = new CardinalityEstimatorConfig(name, 4, 2)
+                .setMergePolicyConfig(new MergePolicyConfig("com.hazelcast.spi.merge.DiscardMergePolicy", 20));
```

Comment:
It should be enough to use `DiscardMergePolicy` here, since the simple classname is also registered.